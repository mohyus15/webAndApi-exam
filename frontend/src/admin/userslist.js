import React, { useState, useEffect } from 'react';
import './UserTable.css'; // Import the CSS file
import UpdateButton from './UpdateUserPage';
import { useParams } from 'react-router-dom';
import { authContext } from '../store/userContext';
import { delete_user } from '../store/types';
import { useContext } from 'react';

const UserTable = () => {
  const [users, setUsers] = useState([]);
  const { userId } = useParams();
  const { dispatch } = useContext(authContext);

  useEffect(() => {
    fetch('http://localhost:8080/api/users')
      .then((response) => response.json())
      .then((data) => setUsers(data))
      .catch((error) => console.error('Error fetching users:', error));
  }, []);

  const handleDelete = async (userIdToDelete) => {
    try {
      const response = await fetch(`http://localhost:8080/api/users/${userIdToDelete}`, {
        method: 'DELETE',
      });
      if (response.ok) {
        const json = await response.json();
        dispatch({ type: delete_user, payload: json });
        setUsers(users.filter(user => user._id !== userIdToDelete));
      } else {
        console.error('Failed to delete user:', response.statusText);
      }
    } catch (error) {
      console.error('Error deleting user:', error);
    }
  };

  const handleUpdate = (userIdToUpdate) => {
    console.log(`Updating user with ID: ${userIdToUpdate}`);
  };

  return (
    <div>
      <h2>User Table</h2>
      <table className="user-table">
        <thead>
          <tr>
            <th>Name</th>
            <th>Email</th>
            <th>Update</th>
            <th>Delete</th>
          </tr>
        </thead>
        <tbody>
          {users.map((user) => (
            <tr key={user._id}>
              <td>{user.name}</td>
              <td>{user.email}</td>
              <td>
                <UpdateButton userId={user._id} onUpdate={handleUpdate} />
              </td>

              <td>
                <button className="delete-button" onClick={() => handleDelete(user._id)}>Delete</button>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default UserTable;
