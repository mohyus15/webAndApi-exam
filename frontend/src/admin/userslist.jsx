import React, { useState, useEffect } from 'react';

const UserTable = () => {
  const [users, setUsers] = useState([]);

  useEffect(() => {
    fetch('http://localhost:8080/api/users')
      .then((response) => response.json())
      .then((data) => setUsers(data))
      .catch((error) => console.error('Error fetching users:', error));
  }, []);

  const handleDelete = (userId) => {
    console.log(`Deleting user with ID: ${userId}`);
  };

  const handleUpdate = (userId) => {
    console.log(`Updating user with ID: ${userId}`);
  };

  return (
    <div>
      <h2>User Table</h2>
      <table>
        <thead>
          <tr>
            <th>ID</th>
            <th>Name</th>
            <th>Email</th>
            <th>Update</th>
            <th>Delete</th>
          </tr>
        </thead>
        <tbody>
          {users.map((user) => (
            <tr key={user.id}>
              <td>{user.id}</td>
              <td>{user.name}</td>
              <td>{user.email}</td>
              <td>
                <button onClick={() => handleUpdate(user.id)}>Update</button>
              </td>
              <td>
                <button onClick={() => handleDelete(user.id)}>Delete</button>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default UserTable;
