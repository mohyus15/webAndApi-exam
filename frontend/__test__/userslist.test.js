import React from 'react';
import { render } from '@testing-library/react';
import userslist from '../src/admin/userslist';

describe('Login Component', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(<userslist/>);
    expect(asFragment()).toMatchSnapshot();
  });
});


