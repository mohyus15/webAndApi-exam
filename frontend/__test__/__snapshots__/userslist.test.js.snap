// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component matches snapshot 1`] = `
<DocumentFragment>
  <userslist />
</DocumentFragment>
`;
